packages/earljs/src/matchers/ObjectWith.ts

```diff
@@ -0,0 +1,37 @@
+import { smartEq } from '../validators/smartEq'
+import { Matcher } from './Base'
+
+/**
+ * Matches an object containing given key-value pairs.
+ */
+export class ObjectWithMatcher<T extends Object> extends Matcher {
+  constructor(private readonly expectedItem: T) {
+    super()
+  }
+
+  check(actualItem: unknown): boolean {
+    if (typeof actualItem !== 'object' || actualItem === null) {
+      return false
+    }
+
+    if ([Set.prototype, Map.prototype, Array.prototype].includes(Object.getPrototypeOf(actualItem))) {
```

Comment:
just use: https://lodash.com/docs/4.17.15#isPlainObject

if not plain object return `false`.